distributed/minGPT-ddp/mingpt/model.py

```diff
@@ -0,0 +1,257 @@
+"""
```

Comment:
mingpt uses an MIT license so we need to retain all the headers https://github.com/karpathy/minGPT/blob/master/LICENSE

I'm a bit more reluctant now to add this example in because we should ideally make it work without relying on external dependencies. But if we do it needs to be crystal clear what we added vs what karpathy wrote

Reply:
Initially I had kept all the files from the minGPT repo as it was a fork. To make maintenance easy, I stripped out the files I'm not using and just kept the minimal set in this PR. What I can do is hyperlink to karpathy's original files for each file that has changed. Wdyt?